[export] turn on replace_set_grad_with_hop_pass in pre_dispatch

ydwu4 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #119915
* #119914
* #119913
* #119810
* #119736
* #119732

This PR turns on replace_set_grad_with_hop_pass for pre_dispatch export. To do that, we need to propagate the meta-data from original submodule to the new higher order op and fix the names of nodes as is required by the _sig_to_specs pass.

ydwu4 committed:
[export] turn on repalce_set_grad_with_hop_pass in pre_dispatch

[ghstack-poisoned]

ydwu4 committed:
Update on "[export] turn on repalce_set_grad_with_hop_pass in pre_dispatch"


This PR turns on replace_set_grad_with_hop_pass for pre_dispatch export.

[ghstack-poisoned]

tugsbayasgalan commented on torch/export/_trace.py:
```diff
@@ -368,6 +368,13 @@ def _export_non_strict(
     if isinstance(mod, torch.fx.GraphModule) and hasattr(mod, "meta"):
         gm.meta.update(mod.meta)
 
+    if pre_dispatch:
```
Could we add a test case for the serialization/deserialization?

ydwu4 replied:
Sure! Thanks for the reminder. Can I do it as a follow-up PR? Want to keep this one short.

tugsbayasgalan replied:
Ok sounds good :)

ydwu4 committed:
Update on "[export] turn on replace_set_grad_with_hop_pass in pre_dispatch"


This PR turns on replace_set_grad_with_hop_pass for pre_dispatch export. To do that, we need to propagate the meta-data from original submodule to the new higher order op and fix the names of nodes as is required by the _sig_to_specs pass.

[ghstack-poisoned]

ydwu4 committed:
Update on "[export] turn on replace_set_grad_with_hop_pass in pre_dispatch"


This PR turns on replace_set_grad_with_hop_pass for pre_dispatch export. To do that, we need to propagate the meta-data from original submodule to the new higher order op and fix the names of nodes as is required by the _sig_to_specs pass.

[ghstack-poisoned]

ydwu4 commented:
@pytorchbot merge

ydwu4 commented:
@pytorchbot merge -f "failure unrelated"
